src/backend/gitporter/read_log_into_database.py

```diff
@@ -0,0 +1,34 @@
+import functools
+from database import DatabaseSession, models as dbm
+from .git_command import parse_git_log
+
+def read_log_into_database(log: str, repository: dbm.Repository, session: DatabaseSession):
+    for parsed_commit in parse_git_log(log):
+        commit = dbm.Commit(
+            committed_at=parsed_commit.committed_at,
+            files_modified=parsed_commit.files_changed,
+            lines_added=parsed_commit.lines_inserted,
+            lines_removed=parsed_commit.lines_deleted,
+            repository=repository,
+            author=get_or_create_author(name=parsed_commit.author_name, email=parsed_commit.email, session=session),
+        )
+        session.add(commit)
+    repository.last_commit_hash = parsed_commit.hash
```

Comment:
```suggestion
    try:
        repository.last_commit_hash = parsed_commit.hash
    except UnboundLocalError: ...
```
wenn der git log leer war, gibt es keine `parsed_commit` variable

Reply:
Wenn der log leer ist, wird None zurückgegeben, kein leerer string. Daher habe ich schon dafür gesorgt, dass das nicht passieren kann.  (auch mit einem leeren Repo getestet)